src/txn/txn_rollback_to_stable.c

```diff
@@ -960,6 +960,88 @@ __rollback_to_stable_btree(WT_SESSION_IMPL *session, wt_timestamp_t rollback_tim
     return (ret);
 }
 
+/*
+ * __rollback_evict_exclusive_apply --
+ *     Apply an eviction helper to all dhandles.
+ */
+static int
+__rollback_evict_exclusive_apply(WT_SESSION_IMPL *session, int (*dhandle_func)(WT_SESSION_IMPL *))
+{
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    const char *uri;
+
+    WT_RET(__wt_metadata_cursor(session, &metadata_cursor));
+
+    while ((ret = metadata_cursor->next(metadata_cursor)) == 0) {
+        WT_ERR(metadata_cursor->get_key(metadata_cursor, &uri));
+
+        /* Evictions of the metadata or history store files won't cause history store operations. */
+        if (WT_STREQ(uri, WT_METAFILE_URI) || WT_STREQ(uri, WT_HS_URI))
+            continue;
+
+        if (!WT_PREFIX_MATCH(uri, "file:"))
+            continue;
+
+        /* Open the dhandle if it's not already open. */
+        ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0);
+
+        if (ret == ENOENT)
+            continue;
+        WT_ERR(ret);
+
+        /* Apply function to dhandle. */
+        WT_ERR((*dhandle_func)(session));
+
+        WT_ERR(__wt_session_release_dhandle(session));
+    }
+    if (ret == WT_NOTFOUND)
+        ret = 0;
+
+err:
+    /* Release any dhandle that we might have. */
+    if (ret != 0)
+        WT_TRET(__wt_session_release_dhandle(session));
+
+    WT_TRET(__wt_metadata_cursor_close(session));
+
+    return (ret);
+}
+
+/*
+ * __rollback_evict_exclusive_on --
+ *     Get exclusive eviction access to any files that may conflict with rollback to stable's
+ *     preliminary check for active transactions.
+ */
+static int
+__rollback_evict_exclusive_on(WT_SESSION_IMPL *session)
+{
+    return (__rollback_evict_exclusive_apply(session, __wt_evict_file_exclusive_on));
+}
+
+/*
+ * __rollback_evict_file_exclusive_off_wrapper --
+ *     A wrapper around the "exclusive off" helper so that we can use it in a similar way to the
+ *     "on" helper.
+ */
+static int
+__rollback_evict_file_exclusive_off_wrapper(WT_SESSION_IMPL *session)
```

Comment:
I didn't clearly understand the need for this wrapper function. The `on` function is directly used why not the `off` also?

Reply:
Because off has `void` return type.